Don't switch executors mid test

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63776
* __->__ #63830
* #63775



It's really not safe to change the executor out from under models that may have
already been partially compiled.

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489/)

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489)

bertmaher committed:
Don't switch executors mid test

It's really not safe to change the executor out from under models that may have
already been partially compiled.

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489/)

[ghstack-poisoned]

bertmaher commented:
For posterity, #63827 contains some of the relevant debugging info that led to this; the "fix" there isn't really correct though.

ZolotukhinM reviewed (approved):
This looks good to me as changing executor mid-flight is not really supported. I don't know if it's critical to disable profiling in those tests, but in case it really is we should be using simple executor (instead of legacy):
```python
torch._C._jit_set_profiling_executor(True)
torch._C._jit_set_profiling_mode(False)
```

bertmaher commented:
Test failures are just because I based this diff on one that disabled fusion of erf, so the tests that check for erf fusion are failing.  I’ll fix that diff and rebase this one in the morning.

bertmaher committed:
Update on "Don't switch executors mid test"




It's really not safe to change the executor out from under models that may have
already been partially compiled.

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489/)

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489)

[ghstack-poisoned]

bertmaher committed:
fix lint on "Don't switch executors mid test"




It's really not safe to change the executor out from under models that may have
already been partially compiled.

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489/)

Differential Revision: [D30504489](https://our.internmc.facebook.com/intern/diff/D30504489)

[ghstack-poisoned]
